### PR TITLE
KR Documentation Update - README - Part 1

### PR DESCRIPTION
**Pull request by kailinr**
## 📝 Summary

Updated README with more explicit language about what MEV-boost is, how it works, and who can run it.

## ⛱ Motivation and Context

Step 1 of docs cleanup effort
---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
 [![Test status](https://github.com/flashbots/mev-boost/workflows/Tests/badge.svg)](https://github.com/flashbots/mev-boost/actions?query=workflow%3A%22Tests%22)
 
-`mev-boost` allows validators in Ethereum proof-of-stake to request blocks from a network of builders.
+## What is MEV-Boost?
 
-This project is part of the Flashbots research towards proposer/builder separation for Maximal Extractable Value (MEV) mitigation. `mev-boost` can connect to relays that aggregate multiple builders. The builders prepare full blocks, optimizing for MEV extraction and fair distribution of the rewards. The Consensus Layer client of the validator proposes the most profitable block received from `mev-boost`.
+`mev-boost` is open source middleware, or a small piece of software, run by validators in order to access a competitive block-building market. MEV-Boost was built by Flashbots as an implementation of [proposer-builder separation (PBS)](https://ethresear.ch/t/proposer-block-builder-separation-friendly-fee-market-designs/9725) for proof-of-stake (PoS) Ethereum.
```
Comment (metachris) [inline README.md:10]:
```suggestion
`mev-boost` is run by validators to access a competitive block-building market. MEV-Boost was built by Flashbots as an implementation of [proposer-builder separation (PBS)](https://ethresear.ch/t/proposer-block-builder-separation-friendly-fee-market-designs/9725) for proof-of-stake (PoS) Ethereum.
```

simplify

Comment (kailinr) [reply README.md:10]:
```suggestion
`mev-boost` is open source middleware run by validators to access a competitive block-building market. MEV-Boost was built by Flashbots as an implementation of [proposer-builder separation (PBS)](https://ethresear.ch/t/proposer-block-builder-separation-friendly-fee-market-designs/9725) for proof-of-stake (PoS) Ethereum.
```

Comment (kailinr) [reply README.md:10]: more explicit use of 'open source' per ofac

**Review by kailinr — COMMENTED** [review]
ofac compromise for line 10?

**Pull request merged**